From 2f7f126168df27069c8ff265f3278bd8c29e1ca2 Mon Sep 17 00:00:00 2001
From: sakinu <sakinu080712345@gmail.com>
Date: Mon, 11 May 2026 05:15:39 +0800
Subject: [PATCH 3/5] use Buildroot ARM toolchain as default CROSS_COMPILE

---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index 61f8019ef..cbd19b284 100644
--- a/Makefile
+++ b/Makefile
@@ -401,6 +401,10 @@ include $(srctree)/scripts/subarch.include
 # Alternatively CROSS_COMPILE can be set in the environment.
 # Default value for CROSS_COMPILE is not to prefix executables
 # Note: Some architectures assign CROSS_COMPILE in their arch/*/Makefile
+ifneq ($(filter arm,$(ARCH)),)
+CROSS_COMPILE ?= ../../host/bin/arm-buildroot-uclinux-uclibcgnueabi-
+endif
+
 ARCH		?= $(SUBARCH)
 
 # Architecture as present in compile.h
-- 
2.54.0

